pkg/vfs/file.go

```diff
@@ -0,0 +1,111 @@
+package vfs
+
+import (
+	"io"
+	"io/fs"
+	"time"
+)
+
+type fileContents struct {
+	node
+	contents []byte
+}
+
+type File struct {
+	*fileContents
+	bytesRead int
+	closed    bool
+}
+
+var (
+	_ fs.File     = (*File)(nil)
+	_ fs.FileInfo = (*File)(nil)
+	_ fs.DirEntry = (*File)(nil)
+)
+
+func (f *File) Name() string {
+	return f.node.name
+}
+
+// Reader is the interface that wraps the basic Read method.
```

Comment:
Copying the doc string from `io.Reader` is a bit excessive :) I think just `Read implements io.Reader` would do.